gradle-baseline-java/src/main/java/com/palantir/baseline/plugins/javaversions/BaselineJavaVersion.java

```diff
@@ -228,7 +228,10 @@ public void execute(Task task) {
             project.getTasks().withType(Checkstyle.class).configureEach(checkstyle -> checkstyle
                     .getJavaLauncher()
                     .set(getJavaLauncher(
-                            rootExtension, baselineConfiguredJavaToolchains, javaToolchainService, target)));
+                            rootExtension,
+                            baselineConfiguredJavaToolchains,
+                            javaToolchainService,
+                            javaCompiler.map(ChosenJavaVersion::of).orElse(target))));
```

Comment:
Needs tests for both `javaCompiler` being set and not. I'd add a comment summarising our discussion about this using `javaCompiler` as an imperfect proxy for what we should run checkstyle with, as it's likely to be higher than the `target` so we can run checkstyle that requires Java 17.